src/main/java/edu/harvard/h2ms/service/report/AbstractReportWorkerComplianceWarning.java

```diff
@@ -37,15 +37,62 @@
 
   long REPORTINGINTERVAL = NotificationFrequency.WEEKLY.seconds;
 
+  public final String COMPLIANCETHRESHOLD = "complianceThreshold";
+
   @Override
   public String getType() {
     return "abstractComplianceWarning";
   }
 
   @Override
-  public String createReport() {
-    List<Event> events = new ArrayList<>();
+  public String createReport(Map<String, String> notificationParameters) {
+
+    double complianceThreshold = 0.80;
+    if (notificationParameters != null && notificationParameters.containsKey(COMPLIANCETHRESHOLD)) {
+      try {
+        complianceThreshold = Double.parseDouble(notificationParameters.get(COMPLIANCETHRESHOLD));
+
+      } catch (NumberFormatException e) {
+        log.error("complianceThreshold has wrong number format.  Falling back to default.");
+      }
+    }
+
+    Map<Question, Map<User, Double>> allComplianceResult = getComplianceResult();
+
+    /** find violators */
+    int violatorCount = 0;
+    Map<Question, Map<User, Double>> allViolators = new HashMap<>();
+    for (Question q : allComplianceResult.keySet()) {
+      Map<User, Double> allQuestionViolators = new HashMap<>();
+      for (User u : allComplianceResult.get(q).keySet()) {
+        Double r = allComplianceResult.get(q).get(u);
+        if (r < complianceThreshold) {
+          allQuestionViolators.put(u, r);
+          violatorCount++;
+        }
+      }
+      allViolators.put(q, allQuestionViolators);
+    }
+
+    String ans = createCsvReport(allViolators);
+    ans =
```

Comment:
Should probably be separate variables. Since it is immediately re-assigned and the first one is just the `csvReport`.